### DISCUSSION TITLE
How to configure the browser version of the Avalonia application

### DISCUSSION THREAD

**Discussion by IVANKIRICHENKO04** [AvaloniaUI/Avalonia#16252] [Q&A]
I have a cross-platform microservice application with an interface developed using Avalonia. The application can connect to a remote API, which I have hosted on a server. Since I use various versions of the API for testing and work, I need to be able to change the API link without recompiling the entire build. In the Windows version of the application, I created an external config where I place the API address, and everything works fine. However, in the browser version, I encountered issues accessing the external config.

I tried to resolve this by implementing various methods in the code, adding and executing JavaScript code, and moving the config content to the index.html file, but none of these approaches worked. What are the options for implementing an external config in the browser version of Avalonia?

**Comment by thevortexcloud** [comment]
You can make the config just a static file on the server.

However if you are trying to access content from another domain, you will probably run into CORS problems, as most browsers/servers block CORS by default.
